File: make-tutorial/dist.R
#!/usr/bin/env Rscript
# Daniel S. Standage
# 3 Sept 2014
# Plot the distances of each observation to a station.

args <- commandArgs(trailingOnly=TRUE)
control.file <- args[1]
treatment.file <- args[2]
output.file <- args[3]

control.data <- read.table(control.file, header=FALSE, col.names=c("Distance"))
treatment.data <- read.table(treatment.file, header=FALSE, col.names=c("Distance"))

control.hist <- hist(control.data$Distance, plot=FALSE)
treatment.hist <- hist(treatment.data$Distance, plot=FALSE)

png(output.file, height=1000, width=1000, res=150)
plot(control.hist$mids, control.hist$counts, col="blue", type="l", main="", xlab="Distance from closest station", ylab="Frequency")
lines(treatment.hist$mids, treatment.hist$counts, col="red")
d <- dev.off()

File: make-tutorial/dist.py
#!/usr/bin/env python
# Daniel S. Standage
# 3 Sept 2014
# Determine the distance between each data point and the closest station.
import sys

class Point:
  def __init__(self, x, y):
    self.x = x
    self.y = y

def distance(p1, p2):
  return ((p1.x - p2.x)**2 + (p2.y - p2.y)**2)**(0.5)

def min_distance(p):
  stations = [Point(0,0), Point(1,1), Point(1,-1), Point(-1,-1), Point(-1,1)]
  distances = []
  for s in stations:
    distances.append(distance(s, p))
  return min(distances)

if __name__ == "__main__":
  for line in sys.stdin:
    x, y = line.rstrip().split("\t")
    p = Point(float(x), float(y))
    d = min_distance(p)
    print "%.4f" % d
